File: c2-chaincode/chaincode-api/chaincode-api-fabric/src/main/java/io/komune/c2/chaincode/api/fabric/config/HasKeystore.java
package io.komune.c2.chaincode.api.fabric.config;

import com.google.common.base.Strings;
import io.komune.c2.chaincode.api.fabric.utils.FileUtils;

import java.io.IOException;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.Properties;



public interface HasKeystore {

    String getKeystore();

    default URL getKeystoreAsUrl(String cryptoBase) throws MalformedURLException {
        return FileUtils.getUrl(cryptoBase, getKeystore());
    }

}

File: c2-chaincode/chaincode-api/chaincode-api-config/src/main/kotlin/io/komune/c2/chaincode/api/config/utils/HasKeystore.kt
package io.komune.c2.chaincode.api.config.utils

import java.net.MalformedURLException
import java.net.URL

interface HasKeystore {
    val keystore: String

    @Throws(MalformedURLException::class)
    fun getKeystoreAsUrl(cryptoBase: String): URL {
        return FileUtils.getUrl(cryptoBase, keystore)
    }
}
